internal/rpcserver/rpcserver.go

```diff
@@ -5878,19 +5911,8 @@ func (s *Server) checkAuth(r *http.Request, require bool) (bool, bool, error) {
 		return false, false, nil
 	}
 
-	mac := make([]byte, 0, sha256.Size)
-	mac = s.authMAC(mac, []byte(authhdr[0]))
-
-	cmp := subtle.ConstantTimeCompare(mac, s.authsha[:])
-	limitcmp := subtle.ConstantTimeCompare(mac, s.limitauthsha[:])
-	if cmp|limitcmp == 0 {
-		// Request's auth doesn't match either user
-		log.Warnf("RPC authentication failure from %s", r.RemoteAddr)
-		return false, false, errors.New("auth failure")
-	}
-
-	isAdmin := cmp == 1
-	return true, isAdmin, nil
+	authed, isAdmin := s.checkAuthHeader(authhdr[0], r.RemoteAddr)
+	return authed, isAdmin, nil
```

Comment:
This is still incorrect.  This is not returning the error when the authentication fails, so the call to it in the non-websocket handler is now incorrect:

```Go
		_, isAdmin, err := s.checkAuth(r, true)
		if err != nil { // This will be `nil` even if the supplied credentials are incorrect....
			jsonAuthFail(w)
			return
		}
```

Reply:
Ah, fully understand now. Sorry. Added more lines in the test.